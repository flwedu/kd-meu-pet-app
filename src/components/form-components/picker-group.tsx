export type PickerOption = {
  id: string;
  alt: string;
  text: string;
};

type PickerProps = {
  spanText: string;
  name: string;
  disabled: boolean;
  options: PickerOption[];
  onChange: (event: any) => void;
};

export function PickerGroup(props: PickerProps) {
  function renderOption(item: PickerOption) {
    return (
      <div key={item.id}>
        <input
          disabled={props.disabled}
          type="radio"
          name={props.name}
          id={item.id}
          value={item.id}
          alt={item.alt}
          onChange={props.onChange}
          required
        />{" "}
        {item.text}
      </div>
    );
  }

  return (
    <div className="form-field">
      <span>{props.spanText}</span>
      <div className="radio-group flex flex-col">
        {props.options.map(renderOption)}
      </div>
    </div>
  );
}
