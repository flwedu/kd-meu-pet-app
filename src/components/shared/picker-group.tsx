type PickerOption = {
  id: string;
  alt: string;
  text: string;
};

type PickerProps = {
  spanText: string;
  name: string;
  disabled: boolean;
  options: PickerOption[];
};

export function PickerGroup(props: PickerProps) {
  function renderOption(option: PickerOption) {
    return (
      <>
        <input
          disabled={props.disabled}
          type="radio"
          name={props.name}
          id={option.id}
          value={option.id}
          alt={option.alt}
        />{" "}
        {option.text}
      </>
    );
  }

  return (
    <>
      <span>{props.spanText}</span>
      <div className="radio-group">{props.options.map(renderOption)}</div>
    </>
  );
}
