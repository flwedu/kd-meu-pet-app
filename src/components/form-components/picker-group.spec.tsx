import { render, fireEvent } from "@testing-library/react";
import { PickerGroup, PickerOption } from "./picker-group";

describe("PickerGroup component tests", () => {
  const props = {
    name: "Tests",
    spanText: "Testing Text",
    options: [],
    onChange: jest.fn(),
    disabled: false,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe("should render the component properly", () => {
    test("With zero children", () => {
      const { container } = render(<PickerGroup {...props} />);
      expect(container).toMatchSnapshot();
    });

    test("With one children", () => {
      const options: PickerOption[] = [
        {
          id: "first",
          alt: "first alt",
          text: "First text",
        },
      ];
      const { container } = render(
        <PickerGroup {...props} options={options} />
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Behavior tests", () => {
    test("Should trigger the onChange event", () => {
      const options: PickerOption[] = [
        {
          id: "first",
          alt: "first alt",
          text: "First text",
        },
      ];
      const { container } = render(
        <PickerGroup {...props} options={options} />
      );
      const input = container.querySelector("input")!;

      fireEvent(input, new MouseEvent("click", { bubbles: true }));

      expect(props.onChange).toHaveBeenCalledTimes(1);
    });
  });
});

export {};
