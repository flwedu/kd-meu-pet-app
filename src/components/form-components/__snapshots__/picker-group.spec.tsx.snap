// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerGroup component tests should render the component properly With one children 1`] = `
<div>
  <div
    class="form-field"
  >
    <span>
      Testing Text
    </span>
    <div
      class="radio-group flex flex-col"
    >
      <div>
        <input
          alt="first alt"
          id="first"
          name="Tests"
          required=""
          type="radio"
          value="first"
        />
         
        First text
      </div>
    </div>
  </div>
</div>
`;

exports[`PickerGroup component tests should render the component properly With zero children 1`] = `
<div>
  <div
    class="form-field"
  >
    <span>
      Testing Text
    </span>
    <div
      class="radio-group flex flex-col"
    />
  </div>
</div>
`;
